### ISSUE TITLE
Criterion of model without attribute `_criterion` overridden by config

### ISSUE THREAD

**Issue by tnw-chatc** [lincc-frameworks/fibad#381]
**Bug report**

I wrote a custom criterion `NTXentLoss` to be used with the Hyrax-decorated model `SimCLR`. In [Line 85](https://github.com/lincc-frameworks/hyrax/blob/f92a99fb6ed697b89247b75471ec3c47082b748c/src/hyrax/models/simclr.py#L85), I specified that the model only uses this custom criterion. I expected that the model would use `NTXentLoss` regardless of the criterion defined in the config file. However, when I run the training, the config file always overrides the criterion, even though it is basically hardcoded into the model code (e.g., default to ordinary cross-entropy loss if not defined, or whatever is defined in the config file).

The bug was found in a PR #379 at commit [104959d](https://github.com/lincc-frameworks/hyrax/pull/379/commits/104959d7ba0b671214ca1724dc197b56710adf0a), which is branched from `model_updates` at commit `5c7a3ea`

Class `SimCLR` does not have attributes `_criterion` nor `_optimizer`, which may cause the bug to happen.

**Before submitting**
Please check the following:

- [ ] I have described the situation in which the bug arose, including what code was executed, information about my environment, and any applicable data others will need to reproduce the problem.
- [ ] I have included available evidence of the unexpected behavior (including error messages, screenshots, and/or plots) as well as a description of what I expected instead.
- [ ] If I have a solution in mind, I have provided an explanation and/or pseudocode and/or task list.

**Comment by aritraghsh09** [comment]
I would suggest running a test where you rename the `NTXentLoss` as `_criterion` 

That will probably take care of this

**Comment by drewoldag** [comment]
This issue was resolved in either Nov or Dec last year. Hyrax now prefers the value defined in the model first, and will fallback to one defines in the config second. 

If it's defined in both places a warning is printed and the model value takes precedence. 

This is true for criterion, optimizer and learning rate scheduler.
